addons/stock/models/stock_quant.py

```diff
@@ -178,7 +178,7 @@ def _update_dict(date_by_quant, key, value):
 
     def _search(self, domain, *args, **kwargs):
         domain = [
-            line if not isinstance(line, (list, tuple)) or not line[0].startswith('lot_properties.')
+            line if not isinstance(line, (list, tuple)) or not (not isinstance(line[0], int) and line[0].startswith('lot_properties.'))
```

Comment:
```suggestion
            line if not isinstance(line, (list, tuple)) or not (isinstance(line[0], str) and line[0].startswith('lot_properties.'))
```
It is good to check the string.

But is it good to use `('active', '=', False),` at the below search?
https://github.com/odoo/odoo/blob/e746182c0a98677a17af8920b681558cdea1495c/addons/stock/models/product.py#L302-L304

Because the actual problem is we archived the warehouses, but above searched location based on warehouse.

@adwid  What do you think about this?

Reply:
So the idea it's just to make it more robust, right ?
Honestly, the `not` make the condition confusing, why not reverse it ?
```py
        domain = [
            ['lot_id', 'any', [line]]
            if isinstance(line, (list, tuple)) and isinstance(line[0], str) and line[0].startswith('lot_properties.')
            else line
            for line in domain
        ]
```